pkg/controller/utils/gcp/utils.go

```diff
@@ -147,37 +147,47 @@ func filterOutPermissions(permList, filterOutList []string) []string {
 // gcpClient creds have sufficient permissions to perform the actions.
 // Will return true/false indicating whether the permissions are sufficient.
 func CheckPermissionsAgainstPermissionList(gcpClient ccgcp.Client, permList []string, logger log.FieldLogger) (bool, error) {
+
 	projectName := gcpClient.GetProjectName()
 
 	filteredPermList := filterOutPermissions(permList, invalidPermsForProjectScopedPermissionsTesting)
 
-	permRequest := &cloudresourcemanager.TestIamPermissionsRequest{
-		Permissions: filteredPermList,
-	}
-	permResponse, err := gcpClient.TestIamPermissions(projectName, permRequest)
-	if err != nil {
-		return false, fmt.Errorf("error testing permissions: %v", err)
+	if len(filteredPermList) == 0 {
+		return true, nil
 	}
 
-	// check that each perm in our list is actually available
-	permMap := map[string]bool{}
-	for _, perm := range permResponse.Permissions {
-		permMap[perm] = true
+	// Split list to only check 100 permissions at a time
+	var chunks [][]string
+	chunkSize := 100
+	permLen := len(filteredPermList)
+	for i := 0; i < permLen; i += chunkSize {
+		end := i + chunkSize
+		if end > permLen {
+			end = permLen
+		}
+		chunks = append(chunks, filteredPermList[i:end])
```

Comment:
You don't need to build a `chunks` slice; just make your request with `Permissions: filteredPermList[I:end]` in this same loop.

Reply:
Ack, fixed